woocommerce/woocommerce-android
# Issue/2191 display product reviews

Fixes #2191 by adding logic to display reviews for a product.

#### Changes
- Added logic to display the reviews count for a product, if reviews are enabled.
- Created a new fragment, module and View Model to handle the changes needed to show only the reviews for a specific product.

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/89783004-d9ac8a80-db33-11ea-8c03-3ac10a840303.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/89783007-db764e00-db33-11ea-86f9-25b8daea208b.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/89783015-ddd8a800-db33-11ea-8f75-4972aba34384.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/89783017-df09d500-db33-11ea-9dc9-225ad581b607.png" width="300"/>

#### Testing
- Click on a product with no reviews enabled.
  - Click on the Product Settings option from the Product Detail menu.
  - Enable reviews for the product and click on `DONE`.
  - Verify that the review count for the product is displayed.
  - Click on the Reviews section.
  - Verify that the approved reviews for the product are displayed correctly.
  - From the web, add a new review on the selected product, and approve it.
  - Pull to refresh the Reviews screen. The new review should appear on top.
- Click on a product with reviews enabled, but no reviews available.
  - Click on the reviews section and verify that an empty view is displayed correctly.
  - From the web, add a new review on the selected product, and approve it.
  - Pull to refresh the Reviews screen. The new review should appear on the screen and the empty view should disappear.
- Click on a product with reviews enabled, and atleast 40 reviews (to test pagination).
  - Click on the reviews section and verify that pagination works as expected.

~**This PR is in draft till this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1659) and this [product type PR](https://github.com/woocommerce/woocommerce-android/pull/2712) can be reviewed and merged.**~

Update release notes:
